src/test/article.spec.js

```diff
@@ -810,3 +850,98 @@ describe('PUT /api/v1/articles/publish?articleId', () => {
     createStub.restore();
   });
 });
+
+describe('GET /api/v1/articles/popular', () => {
+  beforeEach(async () => {
+    await db.Rating.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+    await db.Article.destroy({ cascade: true, truncate: true });
+  });
+  it('should get article by ratings', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const thirdUser = await createUser();
+    const fourthUser = await createUser();
+    const fifthUser = await createUser();
+
+    const firstArticle = await createArticle(
+      await generateArticle({ authorId: firstUser.id })
+    );
+    const secondArticle = await createArticle(
+      await generateArticle({ authorId: firstUser.id })
+    );
+    await rateArticle({
+      articleId: firstArticle.id,
+      userId: secondUser.id,
+      ratings: 5
+    });
+    await rateArticle({
+      articleId: firstArticle.id,
+      userId: thirdUser.id,
+      ratings: 3
+    });
+    await rateArticle({
+      articleId: firstArticle.id,
+      userId: fourthUser.id,
+      ratings: 4
+    });
+    await rateArticle({
+      articleId: secondArticle.id,
+      userId: fifthUser.id,
+      ratings: 1
+    });
+
+    const page = 1;
+    const limit = 3;
+
+    const numberOfRatings = await BaseRepository.findAndCountAll(db.Rating, {
+      where: { articleId: firstArticle.id }
+    });
+
+    const res = await server().get(
+      `${ARTICLES_API}/popular?page=${page}&limit=${limit}`
+    );
+    expect(res.status).to.equal(200);
+    expect(Number(`${res.body.data[0].count_rating}`)).to.equal(
+      Number(`${numberOfRatings.count}`)
+    );
+    expect(
+      Number(Number(`${res.body.data[0].avg_rating}`).toFixed(1))
+    ).to.equal(
+      Number(
+        (
+          numberOfRatings.rows.reduce(
+            (sum, current) => sum + current.ratings,
+            0
+          ) / numberOfRatings.count
+        ).toFixed(1)
+      )
+    );
+    expect(res.body.data[1].id).to.equal(secondArticle.id);
+  });
+
+  it('should return appropriate message if there are no articles', async () => {
+    const page = 1;
+    const limit = 3;
+
+    const res = await server().get(
+      `${ARTICLES_API}/popular?page=${page}&limit=${limit}`
+    );
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.equal('There are no articles at the moment');
+  });
+
+  it.only('should return an error if database error occurs', async () => {
+    const findAllStub = sinon.stub(BaseRepository, 'findByRatingsAndReviews');
+    findAllStub.rejects();
+
+    const page = 1;
+    const limit = 1;
+    const res = await server().get(
+      `${ARTICLES_API}/popular?page=${page}&limit=${limit}`
+    );
+
+    expect(res.status).to.equal(500);
+    findAllStub.restore();
```

Comment:
I think there is a need to provide a message here in the event the server fails.